azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/NetworkSecurityRuleApiLiveTest.java

```diff
@@ -0,0 +1,214 @@
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void getNetworkSecurityRule() {
+      final NetworkSecurityRule rule = createRule();
+      assertNotNull(rule);
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      NetworkSecurityRule result = ruleApi.get(rule.name());
+      assertNotNull(result);
+      assertNotNull(result.etag());
+      assertEquals(result.name(), rule.name());
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void getNetworkSecurityDefaultRule() {
+      String defaultRuleName = "AllowVnetInBound";
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      NetworkSecurityRule result = ruleApi.getDefaultRule(defaultRuleName);
+
+      assertNotNull(result);
+      assertNotNull(result.etag());
+      assertEquals(result.name(), defaultRuleName);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void listNetworkSecurityRules() {
+      final NetworkSecurityRule rule = createRule();
+      assertNotNull(rule);
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      List<NetworkSecurityRule> result = ruleApi.list();
+
+      assertNotNull(result);
+      assertEquals(result.size(), 2);
+
+      boolean rulePresent = false;
+      for (int i = 0; i < result.size(); i++) {
+         if (result.get(i).name().equals(rule.name())) {
+            rulePresent = true;
+            break;
+         }
+      }
+      assertTrue(rulePresent);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void listDefaultSecurityRules() {
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      List<NetworkSecurityRule> result = ruleApi.listDefaultRules();
+
+      assertNotNull(result);
+      assertTrue(result.size() > 0);
+   }
+
+   @Test(groups = "live", dependsOnMethods = {"listNetworkSecurityRules", "listDefaultSecurityRules", "getNetworkSecurityRule"}, alwaysRun = true)
+   public void deleteNetworkSecurityRule() {
+      final NetworkSecurityRule rule = createRule();
+      assertNotNull(rule);
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      URI uri = ruleApi.delete(rule.name());
+      if (uri != null) {
+         assertTrue(uri.toString().contains("api-version"));
+         assertTrue(uri.toString().contains("operationresults"));
+
+         boolean jobDone = Predicates2.retry(new Predicate<URI>() {
+            @Override
+            public boolean apply(URI uri) {
+               return ParseJobStatus.JobStatus.DONE == api.getJobApi().jobStatus(uri);
+            }
+         }, 60 * 2 * 1000 /* 2 minute timeout */).apply(uri);
```

Comment:
@nacx - you had this same advice for @ritazh in PR#664.  Rather than duplicating the work, can we have this addressed as part of the work Rita is doing when she lands that work?  This avoids duplicate efforts.